articles/active-directory/develop/quickstarts/active-directory-windesktop.md

```diff
@@ -0,0 +1,142 @@
+
+#### Step 3: Configure your Visual Studio project
+
+ 1. Open the project in Visual Studio
+ 1. Edit **App.Xaml.cs** and replace the line starting with `private static string ClientId` with the Application ID from the application you just registered:
+
+    ```csharp
+    private static string ClientId = "Enter_the_Application_Id_here";
+    ```
+
+## More Information
+
+### MSAL.NET
+
+MSAL ([Microsoft.Identity.Client](https://www.nuget.org/packages/Microsoft.Identity.Client)) is the library used to sign in users and request tokens used to access an API protected by Microsoft Azure Active Directory. You can install it by running the following command in Visual Studio's *Package Manager Console*:
+
+```powershell
+Install-Package Microsoft.Identity.Client -Pre
+```
+
+### MSAL Initialization
+
+You can add the reference for MSAL by adding the line below:
+
+```csharp
+using Microsoft.Identity.Client;
+```
+
+Then initialize MSAL using the line below:
+
+```csharp
+public static PublicClientApplication PublicClientApp = new PublicClientApplication(ClientId);
+```
+
+> |Where: ||
+> |---------|---------|
+> |ClientId | The Application Id from the application registered in *portal.microsoft.com* |
+
+### Requesting tokens
+
+Msal has two methods used acquire tokens - `AcquireTokenAsync` and `AcquireTokenSilentAsync`:
+
+#### Getting a user token interactively
+
+ Some situations require forcing users interact with Azure Active Directory v2 endpoint via an popup window to either validate their credentials or to give consent. Some examples include:
+
+- The first time users signs-in to the application
+- Users may need to reenter their credentials because the password has expired
+- Your application is requesting access to a resource that the user needs to consent to
+- Two factor authentication is required
+
+```csharp
+authResult = await App.PublicClientApp.AcquireTokenAsync(_scopes);
+```
+
+> |Where:||
+> |---------|---------|
+> |_scopes | Contains the scopes being requested (i.e. `{ "user.read" }` for Microsoft Graph or `{ "api://<Application ID>/access_as_user" }` for custom Web APIs) |
+
+#### Getting a user token silently
+
+You don't want to require user to validate their credentials every time they need to access a resource - most of the time you want token acquisitions and renewal without any user interaction - `AcquireTokenSilentAsync` is the method commonly used to obtain tokens used to access protected resources after the initial `AcquireTokenAsync`:
+
+```csharp
+authResult = await App.PublicClientApp.AcquireTokenSilentAsync(_scopes, App.PublicClientApp.Users.FirstOrDefault());
+```
+
+> |Where: ||
+> |---------|---------|
+> |_scopes | Contains the scopes being requested (i.e. `{ "user.read" }` for Microsoft Graph or `{ "api://<Application ID>/access_as_user" }` for custom Web APIs) |
+ |App.PublicClientApp.Users.FirstOrDefault() | The first user in the cache (MSAL support multiple users in a single app) |
```

Comment:
MSAL support multiple users in a single app=> MSAL **supports** multiple identities in a single app (so that you can write applications like mail clients with several mailboxes for instance)